scripts/process_uploads.py

```diff
@@ -65,15 +65,51 @@ def determine_samples(
 
 def generate_file_list(samples: List[str]):
```

Comment:
Here and elsewhere in this file it would be good to also use type hints for the return values.

Reply:
Thanks, Leo. Really valuable reminder, raised a few issues that needed to be resolved.